src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -43,12 +43,14 @@ def validate_client_parameters(namespace):
     # if account name is specified but no key, attempt to query
     if n.account_name and not n.account_key:
         scf = get_mgmt_service_client(StorageManagementClient)
-        acc_id = next((x for x in scf.storage_accounts.list() if x.name == n.account_name), None).id
+        acc_id = next((x for x in scf.storage_accounts.list() if x.name == n.account_name), None)
```

Comment:
Can we rename `acc_id` to `acc` now as it's actually the account that's returned.